Buildings/Fluid/AirFilters/BaseClasses/MassTransfer.mo

```diff
@@ -2,25 +2,75 @@ within Buildings.Fluid.AirFilters.BaseClasses;
 model MassTransfer
   "Component that sets the trace substance at port_b based on an input trace substance mass flow rate and an input mass transfer efficiency"
   extends Buildings.Fluid.Interfaces.PartialTwoPortInterface;
-  Buildings.Controls.OBC.CDL.Interfaces.RealInput C_inflow[Medium.nC]
-    "Input trace substance rate"
-    annotation (Placement(transformation(extent={{-20,-20},{20,20}},
-      rotation=-90, origin={0,120})));
-  Buildings.Controls.OBC.CDL.Interfaces.RealInput eps(
-    final unit = "1",
-    final min = 0,
-    final max= 1)
+  parameter String substanceName[:] = {"CO2"}
+    "Name of trace substance";
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput eps[size(substanceName,1)](
+    each final unit = "1",
+    each final min = 0,
+    each final max= 1)
     "Mass transfer coefficient"
     annotation (Placement(transformation(extent={{-140,40},{-100,80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput mCon_flow[size(substanceName,1)](
+    each final unit = "kg/s")
+    "Contaminant mass flow rate"
+    annotation (Placement(transformation(extent={{100,40},{140,80}})));
+
+protected
+  parameter Real s1[:,:]= {
```

Comment:
we can add the parameter `nConSub` in the protected section and defined it as `nConSub=size(substanceName,1)`. This will avoid calculating  the `size(substanceName,1)` multiple times.

Reply:
Change it as suggested